var/spack/repos/builtin/packages/siesta/package.py

```diff
@@ -0,0 +1,78 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Siesta(Package):
+    """SIESTA performs electronic structure calculations and ab initio molecular
+       dynamics simulations of molecules and solids."""
+
+    homepage = "https://departments.icmab.es/leem/siesta/"
+    url      = "https://launchpad.net/siesta/4.0/4.0.1/+download/siesta-4.0.1.tar.gz"
+
+    version('4.0.1', '5cb60ce068f2f6e84fa9184ffca94c08')
+
+    depends_on('mpi')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('scalapack')
+
+    phases = ['configure', 'build', 'install']
+
+    def configure(self, spec, prefix):
+        sh = which('sh')
+        configure_args = ['--enable-mpi',
+                          '--with-blas=%s' % spec['blas'].libs.ld_flags,
+                          '--with-lapack=%s' % spec['lapack'].libs.ld_flags,
+                          # need to include BLAS below because Intel MKL's
+                          # BLACS depends on BLAS, otherwise the compiler
+                          # test fails
+                          '--with-blacs=%s %s' % (spec['scalapack'].libs,
```

Comment:
`(spec['scalapack'].libs + spec['blas'].libs).ld_flags`?

Reply:
I tried that, but it just produces gibberish.

Reply:
can you show the result?

Reply:
you would need `'--with-blacs=%s`, of course...

Reply:
This is what `'--with-blacs=%s' % (spec['scalapack'].libs + spec['blas'].libs.ld_flags)` looks like:
```
--with-blacs=/share/apps/spack/opt/spack/linux-centos6-x86_64/intel-15.0.1/intel-mkl-2018.1.163-yxpevoeygg763h75smm2qzf6u5tuhtmp/compilers_and_libraries/linux/mkl/lib/intel64/libmkl_scalapack_lp64.so /share/apps/spack/opt/spack/linux-centos6-x86_64/intel-15.0.1/intel-mkl-2018.1.163-yxpevoeygg763h75smm2qzf6u5tuhtmp/compilers_and_libraries/linux/mkl/lib/intel64/libmkl_blacs_intelmpi_lp64.so - L / s h a r e p c k o t l i n u x 6 8 _ 4 1 5 . 0 m 2 3 y v g 7 q z f d b
```
And it _should_ look like this (i.e. `'--with-blacs=%s' %s % (spec['scalapack'].libs, spec['blas'].libs.ld_flags)`):
```
--with-blacs=/share/apps/spack/opt/spack/linux-centos6-x86_64/intel-15.0.1/intel-mkl-2018.1.163-yxpevoeygg763h75smm2qzf6u5tuhtmp/compilers_and_libraries/linux/mkl/lib/intel64/libmkl_scalapack_lp64.so /share/apps/spack/opt/spack/linux-centos6-x86_64/intel-15.0.1/intel-mkl-2018.1.163-yxpevoeygg763h75smm2qzf6u5tuhtmp/compilers_and_libraries/linux/mkl/lib/intel64/libmkl_blacs_intelmpi_lp64.so -L/share/apps/spack/opt/spack/linux-centos6-x86_64/intel-15.0.1/intel-mkl-2018.1.163-yxpevoeygg763h75smm2qzf6u5tuhtmp/compilers_and_libraries/linux/mkl/lib/intel64 -L/usr/lib64 -lmkl_intel_lp64 -lmkl_core -lmkl_sequential -lpthread -lm -ldl

Reply:
@mkuron but you did not use the code I suggested. It should be 
```
'--with-blacs=%s' % (spec['scalapack'].libs + spec['blas'].libs).ld_flags
```
brackets are kind of important 😄

Reply:
Ah, the little details...